### PR TITLE
Update libmaxminddb to 1.0.4

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

**Comment by tdsmith** [comment]
Test suite is failing on Yosemite and Mountain Lion:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
PASS: bad_pointers_t
PASS: data_types_t
PASS: basic_lookup_t
PASS: data_entry_list_t
PASS: dump_t
PASS: ipv4_start_cache_t
PASS: get_value_t
PASS: get_value_pointer_bug_t
PASS: ipv6_lookup_in_ipv4_t
PASS: metadata_t
PASS: no_map_get_value_t
PASS: read_node_t
PASS: threads_t
PASS: version_t
FAIL: compile_c++_t.pl
PASS: mmdblookup_t.pl
make[4]: Nothing to be done for `all'.
============================================================================
Testsuite summary for libmaxminddb 1.0.4
============================================================================
# TOTAL: 16
# PASS:  15
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See t/test-suite.log
Please report to support@maxmind.com
============================================================================
make[3]: *** [test-suite.log] Error 1
make[2]: *** [check-TESTS] Error 2
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1
```

Does `make check` work in parallel?

**Comment by oschwald** [comment]
Thanks for the heads-up. It looks like it might be a problem with the test itself.

**Comment by tdsmith** [comment]
I saw you made a commit to fix the test; you can apply the commit as a `patch` here if you like.

**Comment by oschwald** [comment]
As far as I can tell, this failure is not related to my change. It looks like a general CI build problem with Mountain Lion.

**Comment by oschwald** [comment]
Any changes required to get this merged? The patch to fix the test is from upstream and will be in the next release.

**Comment by tdsmith** [comment]
Yes, sorry; thanks for the ping. Can you add a comment to that effect above the `patch` declaration with a link to the upstream commit?

File: Library/Formula/libmaxminddb.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Libmaxminddb < Formula
 
   depends_on "geoipupdate" => :optional
 
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/libmaxminddb.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Comment by oschwald** [comment]
I added the comment and squashed the commits yesterday. I am not sure why it is still waiting on the tests.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by oschwald** [comment]
Thanks. It looks like it passed.

File: Library/Formula/libmaxminddb.rb [hunk 2]
```diff
@@ -21,6 +21,11 @@ class Libmaxminddb < Formula
 
   depends_on "geoipupdate" => :optional
 
+  # This patch is from an upstream post-1.0.4 commit and fixes a test failure
+  # on OS X:
+  # https://github.com/maxmind/libmaxminddb/commit/4249537d154b84091e15db09cf00f46d1d19a95f
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/libmaxminddb.rb]: Does https://github.com/maxmind/libmaxminddb/commit/424953.diff apply cleanly as a `patch do` block?

Comment (oschwald) [inline Library/Formula/libmaxminddb.rb]: Yes, I assume you want me to switch to using that directly. Is that correct?

Comment (mikemcquaid) [inline Library/Formula/libmaxminddb.rb]: If it works, yes, thanks.

**Comment by oschwald** [comment]
Updated to use GitHub patch. It looks like the build passed.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `libmaxminddb 1.0.4`.

**Pull request closed without merging**